### PR TITLE
Additional params for index vector [HZAI-57]

### PR DESCRIPTION
**Pull request by shultseva**
added params:
```
private int maxDegree;
private int beamWidth\efConstruction;
private boolean useDeduplication;
```

### CODE REVIEW

File: protocol-definitions/custom/Custom.yaml [hunk 1]
```diff
@@ -1515,6 +1515,18 @@ customTypes:
         type: int
         since: 2.8
         nullable: false
+      - name: maxDegree
+        type: int
+        since: 2.8
+        nullable: true
```
Comment (k-jamroz) [inline protocol-definitions/custom/Custom.yaml:1521]: `FixSizedTypes` fields cannot be nullable (see readme - I was surprised by that)

Comment (shultseva) [reply protocol-definitions/custom/Custom.yaml:1521]: The nullable int value is true or false doesn't impact the generated codec. Additionally, nullable int is already utilized in this file.  So I prefer to leave it as this to maintain alignment with the mono-repo documentation.

Comment (k-jamroz) [reply protocol-definitions/custom/Custom.yaml:1521]: `nullable: true` is ignored here (only for fixed sized types!) and works like `nullable: false`. Do not introduce misleading, ignored setting. If it is used elsewhere for such types, then it is not good either, but with quick search I did not find any such case. Do you have an example of that?

Comment (shultseva) [reply protocol-definitions/custom/Custom.yaml]: 0c1f574e56aea824fd1f55bb529f2f6ea4ddebec

**Review by k-jamroz — APPROVED** [review]

**Review by vbekiaris — APPROVED** [review]

**Pull request merged**